mongo/mongo.go

```diff
@@ -426,14 +432,21 @@ func transformUpdateValue(registry *bsoncodec.Registry, update interface{}, doll
 	}
 }
 
-func transformValue(registry *bsoncodec.Registry, val interface{}) (bsoncore.Value, error) {
+func transformValue(registry *bsoncodec.Registry, val interface{}, mapAllowed bool) (bsoncore.Value, error) {
 	if registry == nil {
 		registry = bson.DefaultRegistry
 	}
 	if val == nil {
 		return bsoncore.Value{}, ErrNilValue
 	}
 
+	if !mapAllowed {
+		refValue := reflect.ValueOf(val)
+		if refValue.Kind() == reflect.Map && refValue.Len() > 1 {
```

Comment:
This reflect logic is cool 😮 Discussed offline, but just want to make sure that third party ordered map implementations like [this one](https://github.com/wk8/go-ordered-map) would not pass `refValue.Kind() == reflect.Map`...

Reply:
Gotcha, thanks for the explanation!